flytekit/core/type_engine.py

```diff
@@ -2052,6 +2052,17 @@ def _check_and_covert_float(lv: Literal) -> float:
     raise TypeTransformerFailedError(f"Cannot convert literal {lv} to float")
 
 
+def _check_and_covert_int(lv: Literal) -> int:
+    if lv.scalar.primitive.integer is not None:
+        return lv.scalar.primitive.integer
+
+    if lv.scalar.primitive.float_value is not None:
+        logger.info(f"Converting literal float {lv.scalar.primitive.float_value} to int, might have precision loss.")
+        return int(lv.scalar.primitive.float_value)
```

Comment:
This is for cases when you input from flyte console, and you use attribute access directly, you have to the `float` to `int`.
Since javascript has only `number`, it can't tell the difference between int and float.

```python
class TrainConfig(BaseModel):
    lr: float = 1e-3
    batch_size: int = 32

@workflow
def wf(cfg: TrainConfig) -> TrainConfig:
    return t_args(a=cfg.lr, batch_size=cfg.batch_size)
```

Reply:
YES, the attribute access works well, it's because javascript pass float to golang, and golang pass float to python.

Reply:
> this should only be a javascript problem. attribute access should work since msgpack preserves float/int even in attribute access correct?

Yes, but when you are accessing a simple type, you have to change the behavior of `SimpleTransformer`.

For Pydantic Transformer, we will use `strict=False` as argument to convert it to right type.
```python
    def from_binary_idl(self, binary_idl_object: Binary, expected_python_type: Type[BaseModel]) -> BaseModel:
        if binary_idl_object.tag == MESSAGEPACK:
            dict_obj = msgpack.loads(binary_idl_object.value)
            python_val = expected_python_type.model_validate(obj=dict_obj, strict=False)
            return python_val
 ```

Reply:
If we can guarantee the console can generate an integer but not float from the input, then we can delete it.

Reply:
how is this going to work though? Do we also do a version check of the backend?

Reply:
After console does the right thing, won't this value be coming in through the binary value instead?  Instead of `lv.scalar.primitive.integer/float`.